### PR TITLE
moved special characters to the end

### PR DESCRIPTION
**Pull request by ejohnson44**

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java [hunk 1]
```diff
@@ -39,14 +41,36 @@ public int compare(SongMetadata songA, SongMetadata songB) {
         int albumComp =  songA.getAlbum().compareTo(songB.getAlbum());
         int titleComp = songA.getTitle().compareTo(songB.getTitle());
 
+        //not the most efficient way of doing things, but it works for now
```
Comment (bryansills) [inline SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java]: can you put a comment about what the code does here?

**Comment by theJenix** [comment]
Doesn't String's built in compareTo function do this?  What does it do with special characters?

**Comment by ejohnson44** [comment]
It puts them at the beginning, which is why the <unkown> artist stuff was coming up to the beginning of our lists, not at the end

**Comment by theJenix** [comment]
I'd like to see if there's a better way to write this comparator...those string matches are really slow, and I don't like the multiple return statements.

**Comment by ejohnson44** [comment]
The multiple returns prevent it from doing unnecessary comparisons - I tried to make it compare things as little as possible so that it wouldn't be too time consuming. 

I could rewrite it to convert to character arrays and manually compare things, but I think that is essentially what the string compare does anyways.

File: SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java [hunk 2]
```diff
@@ -39,14 +41,42 @@ public int compare(SongMetadata songA, SongMetadata songB) {
...
 
+        // not the most efficient way of doing things, but it works for now
+        // if the first character of songA's artist is a special character,
+        // and the first character of songB's artist is a normal letter, songA is ordered
+        // after songB
+        if(!songA.getArtist().substring(0, 1).matches("[a-zA-Z]") && songB.getArtist().substring(0, 1).matches("[a-zA-Z]")){
+            artistComp = 1;
+        }
+        //handles the opposite - A starts with a normal letter and B starts with a special character
+        else if(songA.getArtist().substring(0, 1).matches("[a-zA-Z]") && !songB.getArtist().substring(0, 1).matches("[a-zA-Z]")){
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java]: I dont mind this method of string comparisons for now. My only problem is it looks like repeated code.

**Comment by reidbaker** [comment]
@theJenix What does Elizabeth need to do to get the ok from you?

File: SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java (hunk 2 above)
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/util/AlphabeticalComparator.java]: It's repeated for each element of song, artist, and album, but I can figure out a way to abstract it out to a method

**Pull request merged**